mutant/mutant1/lib/secure_db/document.rb

```diff
@@ -42,7 +42,7 @@ def authorized_users
     #   previously granted access.
     #
     def has_access?(user)
-      (owner == user) ^ authorized_users.include?(user)
+      owner.equal?(user) ^ authorized_users.include?(user)
```

Comment:
This looks odd to me for 2 reasons:

* `#equal?` tests object identity, not equivalency.
* Its very unlikely for production code, you want to use `#equal?` on objects that represent a higher level concept such as an user. Think about loading the user via N access paths (one primary, one via following an association).
* I suggest to instead change the code to `#equal?` you want to add a test showing the code to work with 2 equivalent (but not identical) objects. Hence mutant would than accept `#eql?`.

Reply:
@mbj Thanks for your response. This is an example app however. There is no other way of creating User instance except `User.create(name: 'User1')`. (maybe `#clone`)
Do you think we should add `eql?`\`==` even in this case?

Reply:
@iSarCasm It really "depends". Assuming you can load `User` instances that are equivalent (but have a different object identity) you probably want to use `#eql?` and show that this is the right one via adding a test. It would equal the real world more, and not serve as a bad example.

Locations where you actually *want* `#equal?` are rare. But `#==` is equally bad as it coerces which is unused semantics in ~80% of the cases.